contracts/token/ERC721/ERC721BasicToken.sol

```diff
@@ -4,13 +4,35 @@ import "./ERC721Basic.sol";
 import "./ERC721Receiver.sol";
 import "../../math/SafeMath.sol";
 import "../../AddressUtils.sol";
+import "../../introspection/SupportsInterfaceWithLookup.sol";
 
 
 /**
  * @title ERC721 Non-Fungible Token Standard basic implementation
  * @dev see https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md
  */
-contract ERC721BasicToken is ERC721Basic {
+contract ERC721BasicToken is ERC721Basic, SupportsInterfaceWithLookup {
```

Comment:
Later on there is
```solidity
contract ERC721Token is SupportsInterfaceWithLookup, ERC721BasicToken, ERC721 {
```

`SupportsInterfaceWithLookup` being last in one inheritance list and first in another might bring problems with linearization later on. I would stick to always first or always last. I'm not sure about any of this though. :persevere: Damn C3.